Race condition: parallelized test that affects logging

There seems to be race condition in the test. The problem has been traced back to a function that modifies the logger in tests that are run parallel. The function is `assertLog(t *testing.T, msg string, f func())`. 

### Error Message
```
❯ go test ./...
ok  	github.com/anz-bank/sysl/sysl2/sysl	1.128s
ok  	github.com/anz-bank/sysl/sysl2/sysl/eval	1.512s
ok  	github.com/anz-bank/sysl/sysl2/sysl/exporter	0.171s
?   	github.com/anz-bank/sysl/sysl2/sysl/grammar	[no test files]
ok  	github.com/anz-bank/sysl/sysl2/sysl/importer	(cached)
ok  	github.com/anz-bank/sysl/sysl2/sysl/msg	(cached)
time="2019-10-28T09:57:20+11:00" level=info msg="ff.data = []byte{0x70, 0x61, 0x63, 0x6b, 0x61, 0x67, 0x65}"
--- FAIL: TestSyslParserErrorListenerReportAttemptingFullContext (0.00s)
    error_listener_test.go:48:
        	Error Trace:	error_listener_test.go:48
        	            				error_listener_test.go:82
        	Error:      	Expect "" to match "time="\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(?:Z|[-+]\d\d:\d\d)?" level=info msg="ReportAttemptingFullContext: 42 43\\n"\n"
        	Test:       	TestSyslParserErrorListenerReportAttemptingFullContext
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"ClassId\" (type \"Student\") refers to Field \"Idd\" (type \"Class\") in app \"School\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"ClassId\" (type \"Student\") refers to Field \"Idd\" (type \"Class\") in app \"School\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"field1\" (type \"Output\") refers to type Request in app \"Response\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"ClassId\" (type \"Student\") refers to Field \"Idd\" (type \"Class\") in app \"School\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"MemberId\" (type \"Member\") refers to type Student in app \"SchoolLibrary\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"id\" (type \"Data\") refers to type My Todo App in app \"Browser App\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"id2\" (type \"Data\") refers to type My Todo App in app \"Browser App\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"id3\" (type \"Data\") refers to type My Todo App in app \"Browser App\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"field1\" (type \"Something\") refers to type FooBar in app \"testapp\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Id\" (type \"SecondTable\") refers to type space in app \"AnotherApp\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Id2\" (type \"SecondTable\") refers to type space in app \"AnotherApp\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Id3\" (type \"SecondTable\") refers to type space in app \"AnotherApp\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Id4\" (type \"SecondTable\") refers to type space in app \"AnotherApp\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"nested\" (type \"Response\") refers to type Payload in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"field1\" (type \"TopLevelPayload\") refers to type Model in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"field2\" (type \"TopLevelPayload\") refers to type Client Model in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"bars\" (type \"TopLevelPayload\") refers to type Client Model in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Header\" (type \"Req\") refers to type Header in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Application\" (type \"Req\") refers to type Application in app \"My :: Tuple Model\""
time="2019-10-28T09:57:20+11:00" level=warning msg="Field \"Data\" (type \"Req.Header\") refers to type Tuple Model in app \"My :: Tuple Model\""
FAIL
FAIL	github.com/anz-bank/sysl/sysl2/sysl/parse	1.064s
ok  	github.com/anz-bank/sysl/sysl2/sysl/pbutil	(cached)
ok  	github.com/anz-bank/sysl/sysl2/sysl/roothandler	0.021s
ok  	github.com/anz-bank/sysl/sysl2/sysl/syslutil	(cached)
?   	github.com/anz-bank/sysl/sysl2/sysl/testutil	[no test files]
ok  	github.com/anz-bank/sysl/sysl2/sysl/validate	1.090s
```
### Steps to reproduce the behaviour
`go test ./...`

--- a/sysl2/sysl/parse/error_listener_test.go
+++ b/sysl2/sysl/parse/error_listener_test.go
@@ -12,6 +12,7 @@ import (
 	"github.com/antlr/antlr4/runtime/Go/antlr"
 )
 
+// assertNoLog not to be used in parallel due to logging clashes
 func assertNoLog(t *testing.T, f func()) {
 	var buf bytes.Buffer
 	std := logrus.StandardLogger()
@@ -29,6 +30,7 @@ func assertNoLog(t *testing.T, f func()) {
 	assert.Empty(t, buf.String())
 }
 
+// assertLog not to be used in parallel due to logging clashes
 func assertLog(t *testing.T, msg string, f func()) {
 	var buf bytes.Buffer
 	std := logrus.StandardLogger()
@@ -57,8 +59,6 @@ func (r *dummyRecognizer) GetATN() *antlr.ATN {
 }
 
 func TestSyslParserErrorListenerSyntaxError(t *testing.T) {
-	t.Parallel()
-
 	listener := &SyslParserErrorListener{}
 	recognizer := &dummyRecognizer{antlr.NewBaseRecognizer()}
 	recognizer.SymbolicNames = []string{"some_token", "some_other_token"}
@@ -73,8 +73,6 @@ func TestSyslParserErrorListenerSyntaxError(t *testing.T) {
 }
 
 func TestSyslParserErrorListenerReportAttemptingFullContext(t *testing.T) {
-	t.Parallel()
-
 	listener := &SyslParserErrorListener{}
 	assertNoLog(t, func() {
 		listener.ReportAttemptingFullContext(nil, nil, 42, 43, nil, nil)
@@ -85,8 +83,6 @@ func TestSyslParserErrorListenerReportAttemptingFullContext(t *testing.T) {
 }
 
 func TestSyslParserErrorListenerReportAmbiguity(t *testing.T) {
-	t.Parallel()
-
 	listener := &SyslParserErrorListener{}
 	assertNoLog(t, func() {
 		listener.ReportAmbiguity(nil, nil, 42, 43, false, nil, nil)
@@ -97,8 +93,6 @@ func TestSyslParserErrorListenerReportAmbiguity(t *testing.T) {
 }
 
 func TestSyslParserErrorListenerReportContextSensitivity(t *testing.T) {
-	t.Parallel()
-
 	listener := &SyslParserErrorListener{}
 	assertNoLog(t, func() {
 		listener.ReportContextSensitivity(nil, nil, 42, 43, 0, nil)

